### PR TITLE
[#12048] Prepare for migration testing

### PR DESCRIPTION
**Pull request by FergusMok**
Part of #12048 

1. Add IndexCourseFields and relevant indexes.
2. Added `DataMigrationForCourseEntitySql50Newest` and `DataMigrationForCourseEntitySql50Oldest`

### CODE REVIEW

File: src/client/java/teammates/client/scripts/sql/DataMigrationForCourseEntitySql50Newest.java [hunk 1]
```diff
@@ -0,0 +1,863 @@
...
+ */
+@SuppressWarnings({ "PMD", "deprecation" })
+public class DataMigrationForCourseEntitySql50Newest extends DatastoreClient {
+
+    private static final String BASE_LOG_URI = "src/client/java/teammates/client/scripts/log/";
+
+    private static final int BATCH_SIZE = 100;
+
+    private static final int MAX_BUFFER_SIZE = 1000;
+    private static final AtomicLong NUMBER_TO_MIGRATE = new AtomicLong(50);
```
Comment (FergusMok) [inline src/client/java/teammates/client/scripts/sql/DataMigrationForCourseEntitySql50Newest.java:82]: Important parts to note

File: src/client/java/teammates/client/scripts/sql/DataMigrationForCourseEntitySql50Newest.java [hunk 2]
```diff
@@ -0,0 +1,863 @@
...
+    }
+
+    public static void main(String[] args) {
+        new DataMigrationForCourseEntitySql50Newest().doOperationRemotely();
+    }
+
+    protected Query<Course> getFilterQuery() {
+        return ofy().load().type(teammates.storage.entity.Course.class)
+            .filter("isMigrated", false)
+            .order("-createdAt");
```
Comment (FergusMok) [inline src/client/java/teammates/client/scripts/sql/DataMigrationForCourseEntitySql50Newest.java:122]: Descending time of created

File: src/client/java/teammates/client/scripts/sql/DataMigrationForCourseEntitySql50Newest.java [hunk 3]
```diff
@@ -0,0 +1,863 @@
...
+
+                cursor = iterator.getCursorAfter();
+                savePositionOfCursorToFile(cursor);
+
+                currentOldCourse = iterator.hasNext() ? (Course) iterator.next() : null;
+            }
+            
+            if (numberOfScannedKey == NUMBER_TO_MIGRATE) {
+                shouldContinue = false;
+            }
```
Comment (FergusMok) [inline src/client/java/teammates/client/scripts/sql/DataMigrationForCourseEntitySql50Newest.java:692]: Stops once we hit the number to migrate

**Review by FergusMok — COMMENTED** [review]
Same for Oldest script

File: src/client/java/teammates/client/scripts/sql/DataMigrationForCourseEntitySql50Newest.java (hunk 2 above)
Comment (FergusMok) [reply src/client/java/teammates/client/scripts/sql/DataMigrationForCourseEntitySql50Newest.java:122]: Same for Oldest, where it's ascending

**Review by mingyuanc — APPROVED** [review]
LGTM, thanks for the PR

**Pull request merged**